File: Album/src/main/java/com/hqq/album/activity/AlbumDetailActivity.java
package com.hqq.album.activity;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import com.hqq.album.Adapter.AlbumDetailAdapter;
import com.hqq.album.AppManager;
import com.hqq.album.R;
import com.hqq.album.activity.base.BaseActivity;
import com.hqq.album.common.FunctionKey;
import com.hqq.album.common.FunctionOptions;
import com.hqq.album.common.SelectOptions;
import com.hqq.album.decoration.GridSpacingItemDecoration;
import com.hqq.album.entity.LocalMedia;
import com.hqq.album.utils.AlbumUtils;

import java.util.List;

/**
 * @Author : huangqiqiang
 * @Package : cn.hqq.halbum.activity
 * @FileName :   AlbumDetailActivity
 * @Date : 2018/10/8 0008
 * @Descrive :
 * @Email :
 */
public class AlbumDetailActivity extends BaseActivity implements View.OnClickListener, AlbumDetailAdapter.OnPhotoSelectChangedListener {


    private static final int CODE_CLOSE = 0x9910;
    private RecyclerView mRecyclerView;
    TextView mTvFinish;
    TextView mTvTile;
    AlbumDetailAdapter mAlbumDetailAdapter;

    @Override
    protected void onResume() {
        super.onResume();
        List<LocalMedia> list = SelectOptions.getInstance().getFolderLocalMedia();
        mAlbumDetailAdapter.bindImagesData(list);
        mTvFinish.setText("完成(" + SelectOptions.getInstance().getSelectLocalMedia().size() + "/" + FunctionOptions.getInstance().getMaxSelectNum() + ")");

    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_album_detail);
        initViews();
    }

    private void initViews() {
        mRecyclerView = (RecyclerView) findViewById(R.id.rcv_album_detail);
        mRecyclerView.setHasFixedSize(true);
        mRecyclerView.addItemDecoration(new GridSpacingItemDecoration(4, AlbumUtils.dip2px(this, 2), false));
        mRecyclerView.setLayoutManager(new GridLayoutManager(this, 4));
        mAlbumDetailAdapter = new AlbumDetailAdapter(this, FunctionOptions.getInstance().getMaxSelectNum());
        mRecyclerView.setAdapter(mAlbumDetailAdapter);
        mAlbumDetailAdapter.setOnPhotoSelectChangedListener(this);
        findViewById(R.id.album_back).setOnClickListener(this);
        findViewById(R.id.album_finish).setOnClickListener(this);
        mTvFinish = (TextView) findViewById(R.id.album_finish);
        mTvTile = (TextView) findViewById(R.id.album_title);
        mTvTile.setText(getIntent().getStringExtra(FunctionKey.KEY_FOLDER_NAME));
    }

    @Override
    public void onClick(View view) {
        int i = view.getId();
        if (i == R.id.album_back) {
            AppManager.getAppManager().finishActivity();
        }
        if (i == R.id.album_finish) {
            AppManager.getAppManager().finishAllActivityAndCallBack();
        }
    }

    @Override
    public void onTakePhoto() {// 打开相机

    }

    @Override
    public void onChange(List<LocalMedia> selectImages) {
        mTvFinish.setText("完成(" + selectImages.size() + "/" +  FunctionOptions.getInstance().getMaxSelectNum() + ")");
    }

    @Override
    public void onPictureClick(LocalMedia media, int position) {
        startActivityForResult(new Intent(this, AlbumPreviewActivity.class)
                        .putExtra(FunctionKey.KEY_POSITION, position + 1)
                        .putExtra(FunctionKey.KEY_FOLDER_NAME, getIntent().getStringExtra(FunctionKey.KEY_FOLDER_NAME))
                , CODE_CLOSE);
    }

    @Override
    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        if (requestCode == CODE_CLOSE && resultCode == Activity.RESULT_OK) {
            finish();
        }

    }
}

File: Album/src/main/java/com/hqq/album/activity/AlbumDetailActivity.kt
package com.hqq.album.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.hqq.album.Adapter.AlbumDetailAdapter
import com.hqq.album.Adapter.AlbumDetailAdapter.OnPhotoSelectChangedListener
import com.hqq.album.AppManager
import com.hqq.album.R
import com.hqq.album.common.FunctionKey
import com.hqq.album.common.FunctionOptions
import com.hqq.album.common.SelectOptions
import com.hqq.album.decoration.GridSpacingItemDecoration
import com.hqq.album.entity.LocalMedia
import com.hqq.album.utils.AlbumUtils
import com.hqq.core.ui.base.BaseActivity

/**
 * @Author : huangqiqiang
 * @Package : cn.hqq.halbum.activity
 * @FileName :   AlbumDetailActivity
 * @Date : 2018/10/8 0008
 * @Descrive :
 * @Email :
 */
class AlbumDetailActivity : BaseActivity(), View.OnClickListener, OnPhotoSelectChangedListener {
    private var mRecyclerView: RecyclerView? = null
    var mTvFinish: TextView? = null
    var mTvTile: TextView? = null
    var mAlbumDetailAdapter: AlbumDetailAdapter? = null
    override fun onResume() {
        super.onResume()
        val list = SelectOptions.instance.mFolderLocalMedia
        mAlbumDetailAdapter!!.bindImagesData(list)
        mTvFinish!!.text =
            "完成(" + SelectOptions.instance.selectLocalMedia.size + "/" + FunctionOptions.instance.maxSelectNum + ")"
    }

    override val layoutViewId: Int = R.layout.activity_album_detail
    override fun initConfig() {
        super.initConfig()
        rootViewImpl.iToolBarBuilder.showToolBar = false
    }

    override fun initView() {
        mRecyclerView = findViewById<View>(R.id.rcv_album_detail) as RecyclerView
        mRecyclerView!!.setHasFixedSize(true)
        mRecyclerView!!.addItemDecoration(GridSpacingItemDecoration(4, AlbumUtils.dip2px(this, 2f), false))
        mRecyclerView!!.layoutManager = GridLayoutManager(this, 4)
        mAlbumDetailAdapter = AlbumDetailAdapter(FunctionOptions.instance.maxSelectNum)
        mRecyclerView!!.adapter = mAlbumDetailAdapter
        mAlbumDetailAdapter!!.setOnPhotoSelectChangedListener(this)
        findViewById<View>(R.id.album_back).setOnClickListener(this)
        findViewById<View>(R.id.album_finish).setOnClickListener(this)
        mTvFinish = findViewById<View>(R.id.album_finish) as TextView
        mTvTile = findViewById<View>(R.id.album_title) as TextView
        mTvTile!!.text = intent.getStringExtra(FunctionKey.KEY_FOLDER_NAME)
    }

    override fun onClick(view: View) {
        val i = view.id
        if (i == R.id.album_back) {
            AppManager.appManager?.finishActivity()
        }
        if (i == R.id.album_finish) {
            AppManager.appManager?.finishAllActivityAndCallBack()
        }
    }

    override fun onTakePhoto() { // 打开相机
    }

    override fun onChange(selectImages: List<LocalMedia>?) {
        mTvFinish!!.text = "完成(" + selectImages?.size + "/" + FunctionOptions.instance.maxSelectNum + ")"
    }

    override fun onPictureClick(media: LocalMedia?, position: Int) {
        startActivityForResult(Intent(this, AlbumPreviewActivity::class.java).putExtra(FunctionKey.KEY_POSITION, position + 1)
            .putExtra(FunctionKey.KEY_FOLDER_NAME, intent.getStringExtra(FunctionKey.KEY_FOLDER_NAME)), CODE_CLOSE)
    }

    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)
        if (requestCode == CODE_CLOSE && resultCode == RESULT_OK) {
            finish()
        }
    }

    companion object {
        private const val CODE_CLOSE = 0x9910
    }
}